src/data/CHDatasource.ts

```diff
@@ -66,6 +67,13 @@ export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
     };
   }
 
+  private addColumnTemplateVariable(scoped: ScopedVars, adHocFilters: any): ScopedVars {
+    if (!adHocFilters || !scoped) return scoped;
+    const col = adHocFilters.length > 0 ? adHocFilters[0].key : '';
```

Comment:
hmm not sure if having multiple adhoc filters is a valid scenario, but if so this could be a problem

Reply:
Oh you are totally right. I didnt think about having an adhoc filter like this `SELECT DISTINCT machine_name, machine_group FROM mgbench.logs1`